src/components/application_manager/rpc_plugins/vehicle_info_plugin/include/vehicle_info_plugin/vehicle_info_pending_resumption_handler.h

```diff
@@ -0,0 +1,153 @@
+      CustomVehicleDataManager& custom_vehicle_data_manager);
+
+  void on_event(const app_mngr::event_engine::Event& event) OVERRIDE;
+
+  void HandleResumptionSubscriptionRequest(app_mngr::AppExtension& extension,
+                                           resumption::Subscriber& subscriber,
+                                           app_mngr::Application& app) OVERRIDE;
+  void OnResumptionRevert() OVERRIDE;
+
+  smart_objects::SmartObjectSPtr CreateSubscribeRequestToHMI(
+      const VehicleDataList& subscriptions);
+
+  ns_smart_device_link::ns_smart_objects::SmartObject CreateFakeResponseFromHMI(
+      const std::map<std::string, smart_objects::SmartObject>& subscriptions,
+      const uint32_t fake_corrlation_id);
+
+ private:
+  void ProcessNextPendingResumption(
+      const smart_objects::SmartObject& response_message);
+  void TriggerPendingResumption();
+
+  struct PendingSubscriptionsResumption {
+    PendingSubscriptionsResumption(
+        const uint32_t app_id,
+        const uint32_t corr_id,
+        const VehicleDataList& requested_vehicle_data)
+        : app_id_(app_id)
+        , fake_corr_id_(corr_id)
+        , waiting_for_hmi_response_(false)
+        , requested_vehicle_data_(requested_vehicle_data) {}
+
+    PendingSubscriptionsResumption(const PendingSubscriptionsResumption& copy) =
+        default;
+
+    /**
+     * @brief IsSuccessfullyDone check if SDL restored all required data or not
+     * @return
+     */
+    bool IsSuccessfullyDone() const;
+    bool DataWasRequested(const std::string& vd) const;
+    VehicleDataList NotSubscribedData() const;
+
+    /**
+     * @brief FillSubscriptionResults fill subscription results based on
+     * response message
+     * @param response
+     */
+    void FillSubscriptionResults(const smart_objects::SmartObject& response);
+
+    /**
+     * @brief FillSubscriptionResults fill subscription results based on list of
+     * requested data and list of restored data
+     *
+     */
+    void FillSubscriptionResults();
+
+    void FillRestoredData(const VehicleDataList& successful_subscriptions);
+
+    uint32_t app_id_;
+    uint32_t fake_corr_id_;
+    bool waiting_for_hmi_response_;
+    VehicleDataList requested_vehicle_data_;
+    VehicleDataList restored_vehicle_data_;
+    std::map<std::string, smart_objects::SmartObject> subscription_results_;
+  };
+
+  void SendHMIRequestForNotSubscribed(
+      const PendingSubscriptionsResumption& pending_resumption);
+  void RaiseFinishedPendingResumption(
+      const PendingSubscriptionsResumption& pending_resumption);
+  /**
+   * @brief SubscribeToFakeRequest will create fake subscription for subscriber
+   * ( resumption_data_processor)
+   * Fake request is required only for subscriber subscription.
+   * This request will not be sen't to HMI so it named as fake request.
+   * Fake request contains all data that need to be resumed for the application
+   * When HMI will resopond for any VehicleDara request,
```

Comment:
```suggestion
   * When HMI will respond for any VehicleData request,
```

Reply:
@iCollin fixed in https://github.com/smartdevicelink/sdl_core/pull/3475/commits/11b1c53849491adc3aacbb0f4744a4befabd223f